packages/core/src/render3/interfaces.ts

```diff
@@ -446,6 +437,9 @@ export type InitialInputData = (InitialInputs | null)[];
  */
 export type InitialInputs = string[];
 
+/** The type of the global ngStaticData array. */
+export type NgStaticData = (LNodeStatic | DirectiveDef<any> | null)[];
```

Comment:
As discussed offline:
* Move this to `instructions.ts` to live next to the variable `ngStaticData`
* Explain when this would be `LNodestatic` vs. `DirectiveDef`
* Explain why it is prefixed with `Ng` when nothing else is

@mhevery we also discussed potentially renaming `LNodeStatic` / `ngStaticData` to use the term "flyweight" if you have any opinions on that

Reply:
I like that `data` and `staticData` imply that they go together. (just as field and staticField imply that they are part of the same class) `data` and `flyweight` does not hold that implication. So I would prefer to keep it as is.

Reply:
There's always `data` and `flyweightData`, but ultimately I don't feel very strongly about this.